h/indexer.py

```diff
@@ -13,27 +13,40 @@
 
 log = logging.getLogger(__name__)
 
+SETTING_NEW_INDEX = u'reindex.new_index'
+
 
 def reindex(session, es, request):
     """Reindex all annotations into a new index, and update the alias."""
 
     if get_aliased_index(es) is None:
         raise RuntimeError('cannot reindex if current index is not aliased')
 
+    settings = request.find_service(name='settings')
```

Comment:
In this case I'd suggest that it's probably nicer to actually construct the service yourself using `session` rather than using `request.find_service`. That way we're not introducing another dependency on the `request` object here.